pkg/trait/mount.go

```diff
@@ -0,0 +1,177 @@
+this work for additional information regarding copyright ownership.
+The ASF licenses this file to You under the Apache License, Version 2.0
+(the "License"); you may not use this file except in compliance with
+the License.  You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package trait
+
+import (
+	"fmt"
+	"strings"
+
+	appsv1 "k8s.io/api/apps/v1"
+	"k8s.io/api/batch/v1beta1"
+	corev1 "k8s.io/api/core/v1"
+
+	serving "knative.dev/serving/pkg/apis/serving/v1"
+
+	v1 "github.com/apache/camel-k/pkg/apis/camel/v1"
+	"github.com/apache/camel-k/pkg/util/kubernetes"
+	utilResource "github.com/apache/camel-k/pkg/util/resource"
+)
+
+// The Mount trait can be used to configure volumes mounted on the Integration Pod.
+//
+// +camel-k:trait=mount
+// nolint: tagliatelle
+type mountTrait struct {
+	BaseTrait `property:",squash"`
+	// A list of configuration pointing to configmap/secret. Syntax: [configmap|secret]:name[key], where name represents the resource name and key optionally represents the resource key to be filtered
+	Configs []string `property:"configs" json:"configs,omitempty"`
+	// A list of resources pointing to configmap/secret. Syntax: [configmap|secret]:name[/key][@path], where name represents the resource name, key optionally represents the resource key to be filtered and path represents the destination path
+	Resources []string `property:"resources" json:"resources,omitempty"`
+	// A list of Persistent Volume Claims to be mounted. Syntax: [pvcname:/container/path]
+	Volumes []string `property:"volumes" json:"volumes,omitempty"`
+}
+
+func newMountTrait() Trait {
+	return &mountTrait{
+		// Must follow immediately the container trait
+		BaseTrait: NewBaseTrait("mount", 1610),
+	}
+}
+
+func (t *mountTrait) Configure(e *Environment) (bool, error) {
+	if IsFalse(t.Enabled) {
+		return false, nil
+	}
+
+	if !e.IntegrationInPhase(v1.IntegrationPhaseInitialization) && !e.IntegrationInRunningPhases() {
+		return false, nil
+	}
+
+	// Validate resources and pvcs
+	for _, c := range t.Configs {
+		if !strings.HasPrefix(c, "configmap:") && !strings.HasPrefix(c, "secret:") {
+			return false, fmt.Errorf("unsupported config %s, must be a configmap or secret resource", c)
+		}
+	}
+	for _, r := range t.Resources {
+		if !strings.HasPrefix(r, "configmap:") && !strings.HasPrefix(r, "secret:") {
+			return false, fmt.Errorf("unsupported resource %s, must be a configmap or secret resource", r)
+		}
+	}
+
+	return true, nil
+}
+
+func (t *mountTrait) Apply(e *Environment) error {
+	if e.IntegrationInPhase(v1.IntegrationPhaseInitialization) {
```

Comment:
Could the `Configure` method return `false` then during the `IntegrationPhaseInitialization` phase?